### ISSUE TITLE
Bot on Teams is No Longer Responding in Correct Order

### ISSUE THREAD

**Issue by CharlieEvert** [microsoft/BotFramework-Composer#9531]
## Describe the bug

We have an MS Teams bot that uses Bot Framework composer. It was working properly until March 1st, when it started to encounter issues. It has a flow that used to work which would route users to different types of responses and return SQL query results. It also has a flow that connects to various API-based chatbots. It is built on a resource that once used QNA maker and LUIS even though we are using neither in the current bot. It looks like QNA maker was deprecated so maybe this is part of the issue?

In short, the main issue we are seeing is that suddenly users cannot advance along the chain of dialogs despite them being able to in the past. Could this have to do with autoscaling? How can we even diagnose what is going wrong here?

## Version

We are using V2 or Bot Framework Composer. We are using Dotnet 3.1. 

## Browser

I don't think this is relevant since we're using this on MS Teams. 

- [ ] Electron distribution
- [x] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

## OS

We are using Bot Framework Composer on a mac using Monterrey 12.6.3. The bot itself is hosted on Azure on Windows.

- [x] macOS
- [ ] Windows
- [ ] Ubuntu

## To Reproduce

Enter in text in Teams and send it to the bot, or click one of the multiple-choice options. The appropriate flow will not occur. For instance, a user clicks on "question", they previously were able to enter in a free response question and get an answer but currently, it will shoot the user to "internal data". Then when the user enters in a primary key to receive data, it will randomly teleport them somewhere else.

When testing in the Bot Framework Composer and Emulator, it works fine. When it goes to Teams, we encounter issues.

## Expected behavior

We expect for the user to be able to advance along the flow as they once did. Unfortunately out of the blue, there are issues.

## Screenshots

What we used to have that worked:
<img width="971" alt="Good Flow" src="https://user-images.githubusercontent.com/81777745/223526775-ffe73011-ff4b-46d9-bf9a-8e1381e4e1b4.png">

What we have now that does not work:
<img width="1112" alt="Bad Flow" src="https://user-images.githubusercontent.com/81777745/223526867-d6611c18-34cf-4aa3-8446-aa3314c85d82.png">

## Additional context

We tried redeploying the bot several times and even created new resources to deploy it with. No luck :(

**Comment by anishprasad01** [comment]
Hi @CharlieEvert,

>  Could this have to do with autoscaling?

It's certainly possible. Did you enable cloud state storage before scaling out your bot? 

Once you deploy and are communicating with the hosted bot in Teams, the bot instance that you initially start the conversation flow with (responding with Question) may not be the same instance that handles the next request (responding with Wolfram|Alpha). Since the bot state is stored in the local memory of each instance, each instance cannot understand a conversation started in a different instance. The new instance then interprets it as a new conversation and restarts the flow.

Even if you're still using memory storage, when you test locally everything will work fine since there is only one instance.

**Comment by CharlieEvert** [comment]
Hi @anishprasad01!

Thank you for your response! I do not think we enabled cloud state storage- would this be through Cosmos DB? In the setup JSON we have it left blank as a resource.

It's a pretty weird issue we're having since everything worked fine for a couple of months and then suddenly stopped, even though we made no updates. For one user we did some Regex on their name in teams to get their first name and their name in teams was different from everyone else in the org (so the MS Bot Framework Composer regex errored out) so they cannot advance through any dialog even when we make an update. Is there a way we could reset all of these instances or have users delete chats with bots to reset them?

**Comment by anishprasad01** [comment]
You could try uninstalling the Teams App and reinstalling it. Teams works a little differently compared to the other channels.

Did you have scaling out enabled for those months when everything worked fine? How was the load on the bot during that time?

**Comment by CharlieEvert** [comment]
We made a whole other set of bot resources and we're still having the issue, unfortunately. It does not seem to be tied to auto-scaling since we only had 2 people use the new bot and it has the same issue. We had a 4-instance limit for horizontal scaling, and we were running Standard S1 vertical scaling for that time frame. We increased both and are still having this issue. Here is the load on the bot over the past 90 days, it looks like it may have spiked with a large number of events in early March. 

<img width="780" alt="image" src="https://user-images.githubusercontent.com/81777745/224119045-389e4f73-2020-43b2-ba23-77c919ac35f4.png">
<img width="779" alt="image" src="https://user-images.githubusercontent.com/81777745/224119143-4af262ad-5872-4915-b137-4bf308e6d18d.png">
<img width="785" alt="image" src="https://user-images.githubusercontent.com/81777745/224119485-7c075808-0c7d-4dc9-8bf1-886e6dc3a5b5.png">

We figured out that we actually do have cosmos DB as a resource with our bot. We tried to add blob storage as a rescource but we got an error relating to our org's policy. Could this be it, the blob storage component?
